shinydashboard/lantern/modules/fieldsmodule.R

```diff
@@ -59,23 +61,93 @@ fieldsmodule <- function(
 
   ns <- session$ns
 
-  capstat_extensions_list <- get_capstat_extensions_list(isolate(app_data$capstat_fields()))
+  dstu2 <- c("0.4.0", "0.5.0", "1.0.0", "1.0.1", "1.0.2")
+  stu3 <- c("1.1.0", "1.2.0", "1.4.0", "1.6.0", "1.8.0", "3.0.0", "3.0.1", "3.0.2")
+  r4 <- c("3.2.0", "3.3.0", "3.5.0", "3.5a.0", "4.0.0", "4.0.1")
 
-  output$capstat_fields_text <- renderUI({
-    col <- isolate(app_data$capstat_fields_list()) %>% pull(1)
+  capstat_fields_list <- reactive({
+    res <- isolate(app_data$capstat_fields())
+    version_filter <- FALSE
+
+    req(sel_fhir_version())
+    if (sel_fhir_version() != ui_special_values$ALL_FHIR_VERSIONS) {
+      res <- res %>% filter(fhir_version == sel_fhir_version())
+      version_filter <- TRUE
+    }
+
+    res <- res %>%
+    filter(extension == "false") %>%
+    group_by(field) %>%
+    arrange(fhir_version, .by_group = TRUE)
+
+    # Unknown FHIR versions default to DSTU2
+    res <- res %>% mutate(fhir_version_name = case_when(
+    fhir_version %in% dstu2 ~ "DSTU2",
+    fhir_version %in% stu3 ~ "STU3",
+    fhir_version %in% r4 ~ "R4",
+    TRUE ~ "DSTU2"
+    ))
+
+    res <- res %>% summarise(fhir_version_names = paste(unique(fhir_version_name), collapse = ", "))
+
+    if (version_filter) {
+      res$field_version <- res$field
+    } else {
+      res$field_version <- paste(res$field, " (", res$fhir_version_names, ")", sep = "")
+    }
+
+    res
+})
+
+capstat_extensions_list <- reactive({
```

Comment:
Is there a way to combine some of this work into a function that these can reuse? they look pretty similar besides the check for extension and the field that they're saved to at the end

Reply:
Yeah you are definitely right, they are super similar! I was able to combine them into one function that they both can use